modules/javafx.graphics/src/main/java/javafx/scene/SpotLight.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javafx.scene;
+
+import com.sun.javafx.scene.DirtyBits;
+import com.sun.javafx.scene.NodeHelper;
+import com.sun.javafx.scene.SpotLightHelper;
+import com.sun.javafx.sg.prism.NGNode;
+import com.sun.javafx.sg.prism.NGSpotLight;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.geometry.Point3D;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.PhongMaterial;
+
+/**
+ * A {@code SpotLight} is a {@code PointLight} that radiates light in a cone in a specific direction. The direction is
+ * defined by the {@link #directionProperty() direction} vector property relative to the rotation of the
+ * {@code SpotLight} that is set with the rotation transforms of the node. For example, if the direction vector is
+ * {@code (1, 1, 1)} and the light is not rotated, it will point in the {@code (1, 1, 1)} direction, and if the light is
+ * rotated 90 degrees on the y axis, it will point in the {@code (1, 1, -1)} direction.
+ * <p>
+ * In addition to the attenuation factors defined in {@code PointLight}, the {@code Spotlight}'s intensity is also
+ * affected by 3 factors:
```

Comment:
I think that a big part of the difficulty in explaining what is happening is that we are dancing around the mathematical formulation instead of giving it directly :)
I will update the `SpotLight`'s docs to be more mathematical, but it requires to also update `PointLight`'s docs to be entirely clear. My intention is to update the docs for all the lights and material after we finish adding this one and `DirectionalLight` since it will be possible to do similarities and differences comparisons between them.